### PR TITLE
feat: INFRA-RA-383 Changed container runtime to containerd

### PR DESCRIPTION
**Pull request by Magnitus-**
Changed container runtime to containerd

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -187,4 +186,33 @@ variable "ingress_arguments" {
...
+    //https://docs.tigera.io/calico/latest/networking/configuring/mtu#determine-mtu-size
+    mtu                = number
+    //https://docs.tigera.io/calico/latest/networking/determine-best-networking#networking-options
+    //https://docs.tigera.io/calico/latest/networking/configuring/bgp
+    network_backend    = string
+    encapsulation_mode = string
+  })
+  default = {
+    iptables_backend   = "Legacy"
+    mtu                = 1480
```
Comment (sebastienbonami) [inline variables.tf:214]: This should be 1500 to reflect what the readme says.

File: README.md [hunk 2]
```diff
@@ -42,6 +42,15 @@ The module takes the following input variables:
...
 - **kubespray_image**: Docker image to use for running the kubespray playbooks. The default is **ferlabcrsj/kubespray:2.21.0** which correlates with the value of **kubespray_repo_ref**.
 - **ingress_arguments**: Extra arguments to pass to ingress-nginx (ex: **--enable-ssl-passthrough**).
+- **container_registry_credentials**: Credentials to various container registries to pass to containerd. It should be an array of objects, each with the following keys:
+  - **registry**: Url of the registry
+    **username**: Username to authenticate against the registry
+    **password**: Password to authenticate against the registry
+- **calico**: Parameters to configure the calico cni (https://www.tigera.io/tigera-products/calico/). The defaults are well adapted to our setup. It has the following parameters:
+  - **iptables_backend**: Iptables backend to use. Can be **Legacy**, **NFT** or **Auto**. Defaults to **Legacy**.
+  - **mtu**: Mtu configuration for the virtual interfaces calico will manage. Defaults to 1500. Can be set to 0 in which case calico will get the value from the host's interfaces.
```
Comment (sebastienbonami) [inline README.md]: Ideally put 1500 in bold.

File: README.md [hunk 3]
```diff
@@ -42,6 +42,15 @@ The module takes the following input variables:
 - **kubespray_repo_ref**: Tag or branch to use in the repo before running the kubespray playbooks. The default is the tag **v2.21.0** which is the tag the custom configuration of this repo is adapted to. You may not be successful if you use another tag/branch with different configuration expectations.
 - **kubespray_image**: Docker image to use for running the kubespray playbooks. The default is **ferlabcrsj/kubespray:2.21.0** which correlates with the value of **kubespray_repo_ref**.
 - **ingress_arguments**: Extra arguments to pass to ingress-nginx (ex: **--enable-ssl-passthrough**).
+- **container_registry_credentials**: Credentials to various container registries to pass to containerd. It should be an array of objects, each with the following keys:
+  - **registry**: Url of the registry
+    **username**: Username to authenticate against the registry
+    **password**: Password to authenticate against the registry
```
Comment (sebastienbonami) [inline README.md]: Missing hyphen at the beginning of this line and the line before.

File: README.md [hunk 4]
```diff
@@ -42,6 +42,15 @@ The module takes the following input variables:
...
+- **calico**: Parameters to configure the calico cni (https://www.tigera.io/tigera-products/calico/). The defaults are well adapted to our setup. It has the following parameters:
+  - **iptables_backend**: Iptables backend to use. Can be **Legacy**, **NFT** or **Auto**. Defaults to **Legacy**.
+  - **mtu**: Mtu configuration for the virtual interfaces calico will manage. Defaults to 1500. Can be set to 0 in which case calico will get the value from the host's interfaces.
+  - **network_backend**: Calico network backend to use. Can be **bird**, **vxlan** or "none". Defaults to **bird**.
+  - **encapsulation_mode**: Defines when to use network encapsulation with the chose network backend. Can be **Always**, **CrossSubnet** and **Never**. Defaults to **Always**.
```
Comment (sebastienbonami) [inline README.md]: `or` instead of `and`.

File: variables.tf (hunk 1 above)
Comment (Magnitus-) [reply variables.tf:214]: No, 1480, because you subtract 20 for packet the encapsulation which is the default here. I'll adjust the README instead.

**Review by sebastienbonami — APPROVED** [review]
LGTM

**Pull request merged**